submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+
+//const ITEMS_PER_PAGE = 10; feature pagination
+const usersList = [];
+let usersFilteredList = [];
+const nameInput = document.querySelector('.filter__block input');
+
+const callAPI = async () => {
+    const data = await fetch('https://randomuser.me/api/?results=30&nat=ua');
+    handleErrors(data);
+
+    const { results: users } = await data.json();
+
+    return users;
+};
+
+const handleErrors = (response) => {
+    if (!response.ok) {
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+const getUsers = async() => {
+    try {
+        const users = await callAPI();
+        usersList.push(...users);
+        renderUsers([]);
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+const getOpt = (name) => {
+    return document.querySelector(`.filter__block input[name="${name}"]:checked`);
+}
+
+const mainContainer = document.getElementById('main__content');
+let usersListContainer = document.createElement('div');
+usersListContainer.classList.add('users');
+
+const renderUsers = (pipes, usersFiltered) => {
+    let users = pipes.reduce((list, func) => {
+        if (func.name.startsWith('filter')) { // Filter
+            return list.filter(func);
+        } else if (func.name.startsWith('sort')) { // Sort
+            return list.sort(func)
+        } else { // Do nothing
+            return list;
+        }
+    }, usersList);
+
+    usersFilteredList = users;
+    
+    if (usersFiltered) {
+        users = usersFiltered.map((user) => createUser(user)).join("");
+    } else {
+        users = users.map((user) => createUser(user)).join("");
+    }
```

Comment:
This code doesn't look very dry. Find a way to prepare an array in advance and then write what needs to happen with it once.